cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/cruisecontrol/MockCruiseControlTest.java

```diff
@@ -71,9 +66,8 @@ private void runTest(Vertx vertx, VertxTestContext context, String userTaskID, i
                     response.getJson().getString("Status"),
                     is(CruiseControlUserTaskStatus.COMPLETED.toString()))
             );
-            checkpoint.flag();
             return Future.succeededFuture(response);
-        });
+        }).onComplete(x -> context.completeNow());
```

Comment:
This looks dangerous because you complete the test from a method which is not a test it self.

Reply:
Good point, it should be wrapped in a `context.suceeding()` block.